src/main/scala/db/buckets/bets/BetBucket.scala

```diff
@@ -25,40 +26,79 @@ class BetBucket(cluster: AsyncCluster)(implicit ex: ExecutionContext) extends Ab
     }
   }.toOption
 
-  override def save(data: Bet): Future[Bet] = defaultCollection.insert(entityToId(data), data)(Bet.codec).map(_ => data)
-
-  override def entityToId(data: Bet): String = {
-    s"B:${data.id}"
-  }
-
   override def deleteById(id: String): Future[Unit] = defaultCollection.remove(convertIdToDocId(id)).map(_ => ())
 
   override protected def convertIdToDocId(id: String): String = {
     s"B:${id}"
   }
 
-
   def getBetByUserId(id: String): Future[Seq[Bet]] = {
 
     val query =
       s"""
-         |SELECT *
+         |SELECT b.*
          |FROM `Bets` b
          |WHERE b.userId='$id'""".stripMargin
 
     cluster.query(query)
       .map(_.rowsAs(Bet.codec).getOrElse(Seq()).toSeq)
   }
 
-
-  def getBetByEventIdOneUser(id: String, userId: String): Future[Seq[Bet]] = {
+  def getBetByEventId( eventId: String): Future[Seq[Bet]] = {
     val query =
       s"""
-         |SELECT +
+         |SELECT b.*
          |FROM `Bets` b
-         |WHERE b.eventId='$id' AND b.userId='$userId'""".stripMargin
+         |WHERE b.eventId='$eventId'""".stripMargin
 
     cluster.query(query)
       .map(_.rowsAs(Bet.codec).getOrElse(Seq()).toSeq)
   }
+
+  def getActiveBets: Future[Seq[Bet]] = {
+    for {
+      activeEvents <- eventService.getActiveEventsFromAllPages
+      idActiveEvents = activeEvents.map(_.id.toString).mkString("'", "','", "'")
+      query =
+        s"""
+           |SELECT b.*
+           |FROM `Bets` b
+           |WHERE b.eventId IN [$idActiveEvents]""".stripMargin
+      result <- cluster.query(query).map(_.rowsAs(Bet.codec).getOrElse(Seq()).toSeq)
+    } yield result
+  }
+
+  def createBet(data: Bet, userId: String): Future[Bet] = {
+
+    val newData = data.copy(data.id, data.eventId, userId, data.name, data.price)
+
+    for {
+      checkActiveEventId <- validationActiveEventIdOfBet(newData)
+      checkPrice <- validationBetPriceInGivenRange(newData)
+      bet <- if (checkActiveEventId && checkPrice) defaultCollection.insert(entityToId(newData), newData)(Bet.codec).map(_ => newData) else Future.failed(new RuntimeException(""))
+    } yield bet
+  }
+
+  override def entityToId(data: Bet): String = {
+    s"B:${data.id}"
+  }
+
+  private def validationBetPriceInGivenRange(data: Bet): Future[Boolean] = {
+    if (data.price >= 10 && data.price <= 10000) {
+      Future(true)
+    } else {
+      Future(false)
+    }
+  }
+
+  private def validationActiveEventIdOfBet(data: Bet): Future[Boolean] = {
+    for {
+      event <- eventService.getById(data.eventId)
+      isActive = if (event.get.inPlay.equals(true)) {
```

Comment:
You should avoid doing Optional.get use map() flatMap() getOrElse instead  and again useless if else could be rewrite as`isActive = event.map(_,inPlay).getOrElse(false)`